contracts/libraries/TokenLib.sol

```diff
@@ -278,4 +280,58 @@ library TokenLib {
 
     }
 
+    /**
+     * @notice Validate transfer with TransferManager module if it exists
+     * @dev TransferManager module has a key of 2
+     * @param from sender of transfer
+     * @param to receiver of transfer
+     * @param value value of transfer
+     * @param data data to indicate validation
+     * @param modules Array of addresses for transfer managers
+     * @param modulesToData Mapping of the modules details
+     * @param transfersFrozen whether the transfer are frozen or not.
+     * @return bool
+     */
+    function verifyTransfer(
+        address[] storage modules,
+        mapping(address => ModuleData) storage modulesToData,
+        address from,
+        address to,
+        uint256 value,
+        bytes memory data,
+        bool transfersFrozen
+    ) 
+        public 
+        view
+        returns(bool, byte) 
+    {   
+        if (!transfersFrozen) {
+            bool isInvalid = false;
+            bool isValid = false;
+            bool isForceValid = false;
+            // Use the local variables to avoid the stack too deep error
+            transfersFrozen = false; // bool unarchived = false;
+            byte reasonCode;
+            for (uint256 i = 0; i < modules.length; i++) {
+                if (!modulesToData[modules[i]].isArchived) {
+                    transfersFrozen = true;
+                    (TransferManagerEnums.Result valid, byte reason) = ITransferManager(modules[i]).verifyTransfer(from, to, value, data);
```

Comment:
Don't we need to call the view / non-view version of the function here depending on whether a transfer is actually happening?